rs/replica/src/setup_ic_stack.rs

```diff
@@ -45,9 +48,21 @@ use tokio::sync::{
 /// the message id and height of messages that complete execution.
 const COMPLETED_EXECUTION_MESSAGES_BUFFER_SIZE: usize = 10_000;
 
-/// If enable traffic to `api/v3/call` will be routed to the v3 call handler.
-/// Otherwise requests will be routed to the v2 call handler.
-const ENABLE_V3_CALL_HANDLER: bool = false;
+/// The subnets that can serve synchronous responses to update calls received
+/// on the `/api/v3/.../call`` endpoint.
+const WHITELISTED_SUBNETS_FOR_SYNCHRONOUS_CALL_V3: [&str; 1] =
+    ["snjp4-xlbw4-mnbog-ddwy6-6ckfd-2w5a2-eipqo-7l436-pxqkh-l6fuv-vae"];
+
+fn subnet_is_whitelisted_for_synchronous_call_v3(subnet_id: &SubnetId) -> bool {
+    WHITELISTED_SUBNETS_FOR_SYNCHRONOUS_CALL_V3
+        .iter()
+        .map(|s| {
+            let principal_id = PrincipalId::from_str(s).expect("Whitelist has valid principal ids");
```

Comment:
don't panic, return false

Reply:
I would have preferred to panic at compile time, but `from_str` is unfortunately not const.

If any of the strs are invalid principals, it would be caught during a test. Isn't it better to detect that some of these strings are invalid during tests, instead of just returning false?

Reply:
the risk is low. However, I would prefer not to have any panics when not needed, if we we cannot parse we just fallback to the safest option